disnake/ui/action_row.py

```
@@ -264,19 +353,141 @@ def add_text_input(
+        :class:`ActionRow`:
+            The newly created empty action row, intended for modal components.
+        """
+        return ActionRow[ModalUIComponent]()
+
+    @classmethod
+    def with_message_components(cls) -> ActionRow[MessageUIComponent]:
+        """Create an empty action row meant to store components compatible with
+        :class:`disnake.Message`. Saves the need to import type specifiers to
+        typehint empty action rows.
+
+        .. versionadded:: 2.6
+
+        Returns
+        -------
+        :class:`ActionRow`:
+            The newly created empty action row, intended for message components.
+        """
+        return ActionRow[MessageUIComponent]()
+
+    @classmethod
+    def from_message(cls, message: Message) -> List[ActionRow[MessageUIComponent]]:
+        """Create a list of up to 5 action rows from the components on an existing message.
+
+        This will abide by existing component format on the message, including component
+        ordering and rows. Components will be transformed to UI kit components, such that
+        they can be easily modified and re-sent as action rows.
+
+        .. versionadded:: 2.6
+
+        Parameters
+        ----------
+        message: :class:`disnake.Message`
+            The message from which to extract the components.
+
+        Returns
+        -------
+        List[:class:`disnake.ui.ActionRow`]:
+            The action rows parsed from the components on the message.
+        """
+        rows: List[ActionRow[MessageUIComponent]] = []
+        for row in message.components:
+            rows.append(current_row := ActionRow.with_message_components())
+            for component in row.children:
+                if isinstance(component, ButtonComponent):
+                    current_row.append_item(Button.from_component(component))
+                elif isinstance(component, SelectComponent):
+                    current_row.append_item(Select.from_component(component))
```

Just a suggestion, but this could perhaps use `_component_to_item` if overloads were added to it.
Additionally, do we want to silently ignore unknown component types like right now, or raise an exception instead?

---

Added a strict kwarg for now. Not sure about the `_component_to_item` one as that'd introduce a circular import unless moved. That's something we could consider, though.